crates/ruff_linter/src/rules/refurb/rules/single_item_membership_test.rs

```diff
@@ -129,20 +128,103 @@ fn single_item(expr: &Expr) -> Option<&Expr> {
     }
 }
 
+fn replace_with_comparison(
+    membership_test: MembershipTest,
+    iterable: &Expr,
+    item: &Expr,
+    checker: &Checker,
+) -> Fix {
+    let (locator, source) = (checker.locator(), checker.source());
+    let comment_ranges = checker.comment_ranges();
+
+    let item_range = parenthesized_range(item.into(), iterable.into(), comment_ranges, source)
+        .unwrap_or(item.range());
+    let current_stmt_start = checker.semantic().current_statement().start();
+
+    let replace_op = Edit::range_replacement(
+        membership_test.replacement_op().to_string(),
+        membership_test.range(),
+    );
+    let mut other_edits: SmallVec<[Edit; 2]> = smallvec![];
+
+    let item_in_source = locator.slice(item_range);
+    let replace_iterable_with_item =
+        Edit::range_replacement(item_in_source.to_string(), iterable.range());
+
+    other_edits.push(replace_iterable_with_item);
+
+    let aggregated_comments =
+        merge_to_be_removed_comments(iterable, item_range, current_stmt_start, checker);
+
+    if !aggregated_comments.is_empty() {
+        let move_comments = Edit::insertion(aggregated_comments, current_stmt_start);
+        other_edits.push(move_comments);
+    }
+
+    let applicability = if iterable.is_string_literal_expr() {
+        Applicability::Unsafe
+    } else {
+        Applicability::Safe
+    };
+
+    Fix::applicable_edits(replace_op, other_edits, applicability)
+}
+
+fn merge_to_be_removed_comments(
+    iterable: &Expr,
+    item_range: TextRange,
+    current_stmt_start: TextSize,
+    checker: &Checker,
+) -> String {
+    let (locator, tokens) = (checker.locator(), checker.tokens());
+
+    let mut aggregated_comments = String::new();
+
+    let stmt_indentation_range =
+        TextRange::new(locator.line_start(current_stmt_start), current_stmt_start);
+    let stmt_indentation = locator.slice(stmt_indentation_range);
+    let line_ending = checker.stylist().line_ending().to_string();
+
+    let iterable_start_to_item_start = TextRange::new(iterable.start(), item_range.start());
+    let item_end_to_iterable_end = TextRange::new(item_range.end(), iterable.end());
+
+    tokens
```

Comment:
I'm not sure if we should add more token based fixes. They add a fair amount of complexity.

Reply:
I haven't looked at the fix in detail but my general preference is to accept the removal of comments and instead mark the fix as unsafe if it removes a comment.

Reply:
There's some discussion around comment handling in fixes here: https://github.com/astral-sh/ruff/issues/9790

Reply:
That's fair, although there's a reference to the discussion #9790 that wasn't resolved back then. 

I'm still leaning towards making the fix as unsafe if there's a comment over implementing all this token handling. While impressive, it's just a lot of complexity, especially if we add it to every rule.